modules/olm-overriding-operator-pod-affinity.adoc

```diff
@@ -0,0 +1,272 @@
+kind: Subscription
+metadata:
+  creationTimestamp: '2022-12-21T14:54:47Z'
+  generation: 6
+  labels:
+    operators.coreos.com/openshift-custom-metrics-autoscaler-operator.openshift-keda: ''
+  name: openshift-custom-metrics-autoscaler-operator
+  namespace: openshift-keda
+  resourceVersion: '77887'
+  uid: 9b38d0eb-9c57-4cb6-9905-1c65c21f08ef
+spec:
+  name: my-package
+  source: my-operators
+  sourceNamespace: operator-registries
+  config:
+    affinity:
+      podAntiAffinity: <1>
+        requiredDuringSchedulingIgnoredDuringExecution:
+        - labelSelector:
+            matchExpressions:
+            - key: cpu
+              operator: In
+              values:
+              - high
+          topologyKey: kubernetes.io/hostname
+ ...
+----
+<1> A pod anti-affinity that prevents the Operator's pod from being scheduled on a node that has pods with the `cpu=high` label.
+endif::node[]
+
+.Procedure
+
+To control the placement of an Operator pod, complete the following steps:
+
+. Install the Operator as usual.
+
+. If needed, ensure that your nodes are labeled to properly respond to the affinity.
+
+. Edit the Operator `Subscription` object to add an affinity:
++
+ifndef::pod[]
+[source,terminal]
+----
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  creationTimestamp: '2022-12-21T14:54:47Z'
+  generation: 6
+  labels:
+    operators.coreos.com/openshift-custom-metrics-autoscaler-operator.openshift-keda: ''
+  name: openshift-custom-metrics-autoscaler-operator
+  namespace: openshift-keda
+  resourceVersion: '77887'
+  uid: 9b38d0eb-9c57-4cb6-9905-1c65c21f08ef
+spec:
+  name: my-package
+  source: my-operators
+  sourceNamespace: operator-registries
+  config:
+    affinity: <1>
+      nodeAffinity:
+        requiredDuringSchedulingIgnoredDuringExecution:
+          nodeSelectorTerms:
+            - matchExpressions:
+                - values:
+                    - ip-10-0-185-229.ec2.internal
+                  key: kubernetes.io/hostname
+                  operator: In
+ ...
+----
+ifdef::olm[]
+<1> Add a `nodeAffinity`, `podAffinity`, or `podAntiAffinity`. See the Additional resources section that follows for information about creating the affinity.
+endif::[]
+ifdef::node[]
+<1> Add a `nodeAffinity`.
+endif::[]
+endif::pod[]
+ifdef::pod[]
+[source,terminal]
+----
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  creationTimestamp: '2022-12-21T14:54:47Z'
+  generation: 6
+  labels:
+    operators.coreos.com/openshift-custom-metrics-autoscaler-operator.openshift-keda: ''
+  name: openshift-custom-metrics-autoscaler-operator
+  namespace: openshift-keda
+  resourceVersion: '77887'
+  uid: 9b38d0eb-9c57-4cb6-9905-1c65c21f08ef
+spec:
+  name: my-package
+  source: my-operators
+  sourceNamespace: operator-registries
+  config:
+    affinity:
+      podAntiAffinity: <1>
+        requiredDuringSchedulingIgnoredDuringExecution:
+          nodeSelectorTerms:
```

Comment:
The `nodeSelectorTerms` for the `podAntiAffinity`? I guess it's wrong. And, for Pod `affinity` and `anti-affinity`, an empty topologyKey field is not allowed in both `requiredDuringSchedulingIgnoredDuringExecution` and `preferredDuringSchedulingIgnoredDuringExecution` based on https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/